### PR TITLE
fix(ct): fix exception directly entering checkout payment url

### LINKED ISSUE #5983: Checkout > Payment - Visitting `/checkout/payment` directly from url raises exception
### Expected result
When visitting `/checkout/payment` directly from url when no addresses are yet set, customer should be redirected to previous checkout steps; no exception should be raised

### Actual result
Visitting `/checkout/payment` directly from url when no addresses are yet set results in exception being raised;
e.g.:

```js
TypeError
Cannot read property 'streetName' of null
```

### Steps to reproduce
1. Prerequisite: either shipping nor billing address are yet set
1. Just go to `{{ BASE_URL }}/checkout/payment`

### PR DESCRIPTION
**Pull request by RafalDobrowolski**
## Related Issue

Closes #5983

## How Has This Been Tested?

Automatic e2e tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

#### Changelog
- [x] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [x] I have written test cases for my code
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

#### Code standards
- [x] My code follows the code style of this project.

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

#### Docs
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by lukaszjedrasik — COMMENTED** [review]
Please add changelog: https://docs.vuestorefront.io/v2/contributing/creating-changelog.html

**Comment by Fifciu** [comment]
What if I have an active cart and jump directly to the payment step via URL? I did not fill shipping information, am I able to place an order?

**Comment by RafalDobrowolski** [comment]
> What if I have an active cart and jump directly to the payment step via URL? I did not fill shipping information, am I able to place an order?
I think, there is no possibility place an order, because is available only from the payment step , but if you don't fill out shipping and billing address and try jump directly to the payment step via URL, you redirect to the home page.

**Review by lukaszjedrasik — APPROVED** [review]

File: packages/commercetools/theme/tests/e2e/integration/e2e-checkout-payment-validation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    cy.fixture('test-data/e2e-checkout-payment-validation').then((fixture) => {
+      this.fixtures = {
+        data: fixture
+      };
+    });
+  });
+  it('Should successfully redirect to home page - shipping and billing address are not set', function () {
+    const data = this.fixtures.data[this.test.title];
+    page.checkout.payment.visit();
+    expect(page.home.path).to.be.eq(data.homePath);
```
Comment (igorwojciechowski) [inline packages/commercetools/theme/tests/e2e/integration/e2e-checkout-payment-validation.spec.ts:15]:
```suggestion
    cy.url().should('equal', Cypress.config.baseUrl)
```
Should be something like that. Cuz as it is, we don't compare the actual URL, but hardcoded values.

**Review by igorwojciechowski — CHANGES_REQUESTED** [review]

File: packages/commercetools/theme/tests/e2e/integration/e2e-checkout-payment-validation.spec.ts (hunk 1 above)
Comment (igorwojciechowski) [reply packages/commercetools/theme/tests/e2e/integration/e2e-checkout-payment-validation.spec.ts:15]: Also, test data for this case can be removed after

**Review by igorwojciechowski — APPROVED** [review]

**Review by dawid-ziobro — APPROVED** [review]

**Pull request merged**